rust/connlib/clients/shared/src/control.rs

```diff
@@ -427,15 +427,8 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
                     }
                 });
             }
-            Ok(firezone_tunnel::Event::StopPeer(_)) => {
-                // This should never bubbled up
-                // TODO: we might want to segregate events further
-            }
-            Ok(firezone_tunnel::Event::SendPacket(_)) => {
-                unimplemented!("Handled internally");
-            }
             Err(e) => {
-                tracing::error!("Tunnel failed: {e:#?}");
+                tracing::warn!("Tunnel error: {e}");
```

Comment:
One significant change with this PR is that I am liberally using `?` inside the `poll_next_event` function because we only print the error anyway. This happens occasionally, like trying to send a packet to a not (yet) connected peer. Hence, I downgraded this to a warning.

I think this is overall better for now. We need carefully think about, which errors to abort the tunnel on and that should be captured in the type-system. Until then, we can just `?` everything inside the tunnel because we will never abort it.